src/components/cylc/Graph.vue

```diff
@@ -0,0 +1,1633 @@
+            classes: ''
+          }
+          if (has(node, 'task') && !isEmpty(node.task) && has(node.task, 'name') && !isEmpty(node.task.name)) {
+            nodeObj.data.label = node.task.name
+          }
+          has(node, 'id') && !isEmpty(node.id) ? nodeObj.data.id = node.id : console.warn('workflowUpdated - node id is empty')
+          has(node, 'state') && !isEmpty(node.state) ? nodeObj.data.state = node.state : nodeObj.state = 'undefined'
+          has(node, 'status') && !isEmpty(node.status) ? nodeObj.data.status = node.status : nodeObj.status = 'undefined'
+          if (has(node, 'runpercent') && !isEmpty(node.runpercent) && (parseInt(node.runpercent) > 0)) {
+            nodeObj.data.runpercent = parseInt(node.runpercent)
+            nodeObj.running = nodeObj.data.runpercent
+            todo = 100 - parseInt(nodeObj.data.runpercent)
+            nodeObj.data.todo = todo
+          }
+          if (has(node, 'runpercent') && !isEmpty(node.runpercent) && (parseInt(node.runpercent) > 0)) {
+            nodeObj.data.runpercent = parseInt(node.runpercent)
+            nodeObj.running = nodeObj.data.runpercent
+            todo = (100 - parseInt(node.runpercent))
+            nodeObj.data.todo = todo
+          }
+          if (has(node.parent, 'id') && !isEmpty(node.parent.id)) {
+            nodeObj.data.parent = node.parent.id
+          }
+          if (has(node, 'cyclePoint') && !isEmpty(node.cyclePoint)) {
+            nodeObj.data.cyclepoint = node.cyclePoint
+          }
+          if (has(node, 'jobs') && !isEmpty(node.jobs)) {
+            jobsArray = this.getJobs(node.jobs)
+            nodeObj.data.jobsarray = jobsArray
+            const joblength = node.jobs.length
+            nodeObj.data.jobs = String(node.jobs.length)
+            nodeObj.data.submittedtime = jobsArray[0].submittedTime
+            nodeObj.data.starttime = jobsArray[0].startedTime
+            nodeObj.data.finishtime = jobsArray[joblength - 1].finishedTime
+          }
+          nodesArray.push(nodeObj)
+        })
+        // console.debug('NODES ::: ', JSON.stringify(nodesArray))
+        return nodesArray
+      } catch (error) {
+        console.error('getNodes error: ', error)
+      }
+    },
+
+    getEdges (edges) {
+      try {
+        const edgesArray = []
+        let edgeObj = {}
+        each(edges, (edge, key) => {
+          edgeObj = {
+            data: {
+              id: '',
+              source: '',
+              target: '',
+              label: '',
+              owner: '',
+              name: ''
+            },
+            position: {
+
+            },
+            group: 'edges',
+            removed: false,
+            selected: false,
+            selectable: true,
+            locked: false,
+            grabbable: true,
+            classes: ''
+          }
+          has(edge, 'id') && !isEmpty(edge.id) ? edgeObj.data.id = edge.id : console.debug('workflowUpdated - edge id is empty')
```

Comment:
These functions workflowUpdated() getEdges() and GetNodes() construct the required cytoscape data format and are a workaround until the graph is supplied with flat data, then they can be removed and the data supplied to the updateGraph() function. This will really speed performance back up again.